### ISSUE TITLE
Add a category option and check if the rules added as overrides are not of that category

### ISSUE THREAD

**Issue by joaopedrodcf** [oxc-project/oxlint-migrate#423]
Hello,

Was using the tool to generate an oxlint configuration file but it creates so many rules that is a bit difficult to manage the configuration because correctness is off but I actually wanted to force it to be true.

This way I could remove a bunch of the rules it generates and only have the ones from other categories

```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [
    "unicorn",
    "import"
  ],
  "categories": {
    "correctness": "off" // I wanted this to be error or warn 
  },
  "env": {
    "builtin": true,
    "browser": true,
    "es2024": true,
    "node": true
  },
  {
  "no-unexpected-multiline": "error", // supecious I would like to keep
  "constructor-super": "error", // this one is part of corrtness
   ... 
}
```

This is what I was thinking we could have in the CLI
```
❯ npx @oxlint/migrate eslint.config.mjs --js-plugins=false category=correctness
```

Or even better if we could define a list of categories and if they are warn, error etc

**Comment by Sysix** [comment]
Did you try using `--merge` flag after enabling the categories? The flag is not perfect integrated, but it should show you a little git diff for removed rules 🤞

**Comment by nihalgonsalves** [comment]
I took the rules metadata from https://github.com/oxc-project/website/blob/e7e5e8cde28c141eab0a0ff830a4da45c5bf2530/.vitepress/data/rules.json and made a small script that can prune rules from enabled categories. It isn't aware of anything but `error` and rules that have no custom config, and isn't aware of aliases (e.g. `@typescript-eslint/`, so you'll have to find and replace first – #413 is related) but works to clean up a lot of things: https://gist.github.com/nihalgonsalves/f23c5b548cb5df4bf4a82084c22f3a2c

would be happy to add (a better version of) this to this project if that's something we want!

- example config pre-pruning: https://github.com/nihalgonsalves/esconfig/blob/5e01b76dd7b94b9d08cd2957b7097785ded19bc3/.oxlintrc.json
- after: https://github.com/nihalgonsalves/esconfig/blob/4cbf8860b69a8e2ad929a7d8268a18ff354cc464/.oxlintrc.json

**Comment by joaopedrodcf** [comment]
I think I tried --merge and it doens't does what I intended @Sysix

**Comment by joaopedrodcf** [comment]
Also update the issue to be more clear what I meant 🙏

**Comment by joaopedrodcf** [comment]
That really helped @nihalgonsalves , I took the liberty to update your script a bit https://gist.github.com/joaopedrodcf/20ffa0dc90838986d52a8a89e23d7112

<img width="811" height="681" alt="Image" src="https://github.com/user-attachments/assets/11de86e6-c89e-481c-a396-bb9e194caa0a" />
